config/schema.json

```diff
@@ -15,6 +15,38 @@
       "type": "string",
       "minLength": 1
     },
+    "consulHTTPAddr": {
+      "description": "The HTTP(S) URL of the Consul server. Required when `authMethod.enabled` is set",
+      "type": ["string", "null"]
+    },
+    "consulCACertFile": {
+      "description": "The file path of the Consul server CA certificate.",
+      "type": ["string", "null"]
+    },
+    "consulLogin": {
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "description": "Enable login to Consul's AWS IAM auth method to obtain an ACL token. This requires the auth method to be configured on the Consul server, and the ECS task role must be trusted by the auth method. After login, the token is written to the file `<bootstrapDir>/service-token`.",
+          "type": "boolean"
+        },
+        "method": {
+          "description": "The name of Consul auth method. This is passed as the `-method` option to the `consul login` command. Defaults to `iam-ecs-service-token`.",
+          "type": ["string", "null"]
+        },
+        "noIncludeEntity": {
```

Comment:
Yeah, I agree. I opted for this because it was awkward in Golang to have it default to `true`.

At one point I had something like the following. You need a `*bool` to distinguish if the field is present or not in the JSON, but you need an exported field in the struct for the `json` package.

```go
// ConsulLogin configures login options for the Consul IAM auth method.
type ConsulLogin struct {
    Enabled bool   `json:"enabled"`
    Method  string `json:"method"`
    // Exported field is needed for JSON unmarshalling.
    // But, this defaults to true so you must use c.IncludeEntity().
    RawIncludeEntity *bool    `json:"includeEntity"`
    ExtraLoginFlags  []string `json:"extraLoginFlags"`
}

func (c *ConsulLogin) IncludeEntity() bool {
    if c.RawIncludeEntity == nil {
        // not set - default to true
        return true
    }
    return *c.RawIncludeEntity
}
```

Thoughts?

Reply:
I switched this to `includeEntity` with default false, and used a custom `UnmarshalJSON` to set the default.